src/libsyntax/parse/parser.rs

```diff
@@ -2663,7 +2665,7 @@ impl<'a> Parser<'a> {
     }
 
     pub fn check_unknown_macro_variable(&mut self) {
-        if self.quote_depth == 0 {
+        if self.quote_depth == 0 && !self.parsing_token_tree {
```

Comment:
Specifically, it caused problems with parsing the left hand side of a `macro_rules!` body. I didn't look into it in too much detail since I wanted to do this PR as simply as possible.